openpype/plugins/publish/extract_slate_global.py

```diff
@@ -0,0 +1,538 @@
+
+        repre_match = None
+
+        # loop through representations and generate a slate frame for each
+        for repre in instance.data["representations"]:
+            if repre["name"] in repre_ignore_list:
+                self.log.debug(
+                    "Representation '{}' was ignored.".format(repre["name"])
+                )
+                continue
+
+            # check if repre is a sequence
+            is_sequence = False
+            check_file = repre["files"]
+            if isinstance(check_file, list):
+                check_file = check_file[0]
+                is_sequence = True
+
+            file_path = os.path.normpath(os.path.join(repre["stagingDir"], check_file))
+
+            # if representation is a sequence render out a slate before first frame
+            if is_sequence:
+                filename, _frame, ext = check_file.split(".")
+                frame_start = int(repre["frameStart"]) - 1
+                frame_end = len(repre["files"]) + frame_start
+                output_name = "{}.{}.{}".format(
+                    filename,
+                    str(frame_start).zfill(int(common_data["frame_padding"])),
+                    ext,
+                )
+                thumbnail_path = os.path.join(
+                    repre["stagingDir"],
+                    "{}_slate_thumb.png".format(repre["name"]),
+                ).replace("\\", "/")
+                slate_creator.render_image_oiio(
+                    file_path.replace("\\", "/"), thumbnail_path
+                )
+                repre_match = instance.data["family"]
+
+            else:  # else find matching tags and transfer
+                frame_start = int(repre["frameStart"])
+                frame_end = int(repre["frameEnd"])
+                thumbnail_path = repre_thumbnail_path
+                output_name = "{}_slate_temp.png".format(repre["name"])
+                for tag in repre["tags"]:
+                    for profile in slate_data["slate_profiles"]:
+                        if tag in profile["families"]:
+                            repre_match = tag
+
+            # add timecode to oiio output args
+            timecode = slate_creator.get_timecode_oiio(
+                file_path, timecode_frame=int(repre["frameStart"])
+            )
+            oiio_profile = {}
+            for profile in slate_data["slate_profiles"]:
+                if repre_match in profile["families"]:
+                    oiio_profile = profile
+                else:
+                    oiio_profile = {
+                        "families": [],
+                        "hosts": [],
+                        "oiio_args": {"input": [], "output": []},
+                    }
+                oiio_profile["oiio_args"]["output"].extend(
+                    [
+                        "--attrib:type=timecode",
+                        "smpte:TimeCode",
+                        '"{}"'.format(timecode),
+                    ]
+                )
```

Comment:
1. Should this break on the first match?
2. Since we have an `else` fallback doesn't it just always take the default if it doesn't have a match?

How about:

```suggestion
            # Default
            oiio_profile = {
                "families": [],
                "hosts": [],
                "oiio_args": {"input": [], "output": []},
            }
            for profile in slate_data["slate_profiles"]:
                if repre_match in profile["families"]:
                    # Use profile matching defaults
                    oiio_profile = profile
                    break
            
            # Add timecode attribute
            oiio_profile["oiio_args"]["output"].extend(
                [
                    "--attrib:type=timecode",
                    "smpte:TimeCode",
                    '"{}"'.format(timecode),
                ]
            )
```

Also, I feel like this behaves differently than other "profiles" with profile filtering in OpenPype. Should it behave more similar and use the regular filter_profiles function?

Reply:
Hard to tell what the intended behavior is there. I'd say if this is about matching families/profiles to instead rely on regular profiles filtering that OpenPype already has dedicated functions for just so it's using the same profiles filtering system as is used in other areas for consistency. That function instead takes the "best match" instead of the first.